### PR TITLE
Machine ID: Add guides to the Enroll Integration page

### LINKED ISSUE #23172: Machine ID: Teleport Discover integration

### PR DESCRIPTION
**Pull request by strideynet**
Adds tiles to Enroll Integration that link to Machine ID docs and fire off an analytics event.

image
image

Closes https://github.com/gravitational/teleport/issues/23172

### CODE REVIEW

File: web/packages/design/src/SVGIcon/GitLab.tsx
Comment (strideynet) [inline web/packages/design/src/SVGIcon/GitLab.tsx:1]: This appears too small :/

File: web/packages/design/src/SVGIcon/CircleCI.tsx
Comment (strideynet) [inline web/packages/design/src/SVGIcon/CircleCI.tsx:1]: Appears slightly cut off

File: web/packages/teleport/src/Integrations/Enroll/common.tsx [hunk 1]
```diff
@@ -57,3 +64,84 @@ export const NoCodeIntegrationDescription = () => (
...
+    icon: JSX.Element
+  }
+  const tiles: tile[] = [
+    // TODO: Emit an event or pass these through a /r/ url to redirect to a
+    // URL with tracking.
+    // Can we emit events in OSS and have them go nowhere?? Is it best to use
+    // a /r/ url and have support on both Enterprise and OSS.
+    {
+      title: 'GitHub Actions',
+      link: 'https://goteleport.com/docs/machine-id/guides/github-actions/',
```
Comment (strideynet) [inline web/packages/teleport/src/Integrations/Enroll/common.tsx:81]: Keeping track of some of my owns thoughts:

Do we want to link these through a `/r/` redirect and then attach UTM details ? Or do we want to not do that, and emit an event when on Enterprise/Cloud ? Can I emit an event in OSS and just have it sinkholed or do I need to DI in the emitter in `e` and push in a nop implementation in OSS.

**Comment by strideynet** [comment]
DO NOT MERGE UNTIL https://github.com/gravitational/cloud/pull/5157

**Comment by gzdunek** [comment]
It seems to me that we use color logos if possible. If so, shouldn't we use the full color version for GitLab? https://about.gitlab.com/press/press-kit 
Also, GHA has its own logo, right? https://www.google.com/search?q=github+actions+logo&tbm=isch&client=safari&hl=en&sa=X&ved=2ahUKEwiU3Nvzqvz_AhVPEBAIHe1VAz0QrNwCKAB6BQgBEJ0C&biw=2264&bih=1346

File: lib/usagereporter/web/userevent.go [hunk 2]
```diff
@@ -247,10 +248,12 @@ func ConvertUserEventRequestToUsageEvent(req CreateUserEventRequest) (*usageeven
 			return nil, trace.BadParameter("eventData is invalid: %v", err)
 		}
 
+		fmt.Println("Json unmarshalled for start event")
```
[the remark this answers is not in the document]
Comment (strideynet) [reply lib/usagereporter/web/userevent.go:251]: Good catch

File: web/packages/teleport/src/Integrations/Enroll/common.tsx [hunk 3]
```diff
@@ -57,3 +72,93 @@ export const NoCodeIntegrationDescription = () => (
     </Text>
   </Box>
 );
+
+export const MachineIDIntegrationSection = () => {
```
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx:76]: Yeah - that's a nice idea.

File: web/packages/teleport/src/Integrations/Enroll/common.tsx [hunk 4]
```diff
@@ -57,3 +72,93 @@ export const NoCodeIntegrationDescription = () => (
...
+        </Text>
+        <Text typography="body1">
+          Set up Teleport Machine ID to allow CI/CD workflows and other machines
+          to access resources protected by Teleport.
+        </Text>
+      </Box>
+      <Flex mb={2} gap={3}>
+        {tiles.map((t: tile) => {
+          return (
+            <IntegrationTile {...propsForTile(t)}>
```
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx:153]: No particular reason - it's the first React I've written in a year so I'm a little rusty. I'll do that.

File: web/packages/teleport/src/Integrations/Enroll/common.tsx [hunk 5]
```diff
@@ -57,3 +72,93 @@ export const NoCodeIntegrationDescription = () => (
...
   </Box>
 );
+
+export const MachineIDIntegrationSection = () => {
+  interface tile {
+    title: string;
+    link: string;
+    icon: JSX.Element;
+    kind: IntegrationEnrollKind;
+  }
```
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx:82]: Neither did I - agreed. I'll pull it out when I move `tiles` out and make this a seperate file.

File: web/packages/teleport/src/Integrations/Enroll/common.tsx [hunk 6]
```diff
@@ -57,3 +72,93 @@ export const NoCodeIntegrationDescription = () => (
...
+      icon: <AnsibleIcon size={80} />,
+      kind: IntegrationEnrollKind.MachineIDAnsible,
+    },
+    {
+      title: 'Generic',
+      link: 'https://goteleport.com/docs/machine-id/getting-started/',
+      icon: <ServersIcon size={80} />,
+      kind: IntegrationEnrollKind.MachineID,
+    },
+  ];
```
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx]: 904df3cd58

File: web/packages/teleport/src/Integrations/Enroll/common.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx]: 904df3cd58

File: web/packages/teleport/src/Integrations/Enroll/common.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx]: 904df3cd58

File: web/packages/teleport/src/Integrations/Enroll/common.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/common.tsx]: 904df3cd58

File: lib/usagereporter/web/userevent.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply lib/usagereporter/web/userevent.go]: [ca4f4a1](https://github.com/gravitational/teleport/pull/28646/commits/ca4f4a184a52464b4aa49ed00a39f53434c8e9c9)

**Comment by strideynet** [comment]
image
Updated to use GitLabs coloured icon mark.

File: web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+} from 'design/SVGIcon';
+import { Box, Flex, Link as ExternalLink, Text } from 'design';
+import React from 'react';
+
+import {
+  IntegrationEnrollEvent,
+  IntegrationEnrollKind,
+  userEventService,
+} from 'teleport/services/userEvent';
+import { IntegrationTile } from 'teleport/Integrations/Enroll/common';
```
Comment (gzdunek) [inline web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx:33]: We import from the same directory, so it is better to use relative import.

```suggestion
import { IntegrationTile } from './common';
```

File: web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+  },
+  {
+    title: 'Generic',
+    link: 'https://goteleport.com/docs/machine-id/getting-started/',
+    icon: <ServersIcon size={80} />,
+    kind: IntegrationEnrollKind.MachineID,
+  },
+];
+
+const MachineIDIntegrationSection = () => {
```
Comment (gzdunek) [inline web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx:81]: Please use named export (if you want to know why named exports are better: https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad)

```suggestion
export const MachineIDIntegrationSection = () => {
```

**Review by gzdunek — APPROVED** [review]
With two small comments.

File: web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx (hunk 8 above)
Comment (strideynet) [reply web/packages/teleport/src/Integrations/Enroll/MachineIDIntegrationSection.tsx]: 2ef44fb200

**Comment by strideynet** [comment]
Checked protos against the cloud repo master branch - no changes to make.

**Review by ryanclark — APPROVED** [review]

**Pull request merged**